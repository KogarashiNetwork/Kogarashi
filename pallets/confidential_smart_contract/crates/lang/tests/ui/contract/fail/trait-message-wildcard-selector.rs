mod foo {
    use ink_lang as ink;

    #[ink::trait_definition]
    pub trait TraitDefinition {
        #[ink(message, selector = _)]
        fn message(&self);
    }
}

use ink_lang as ink;

#[ink::contract]
pub mod contract {
    use super::foo::TraitDefinition;

    #[ink(storage)]
    pub struct Contract {}

    impl Contract {
        #[ink(constructor)]
        pub fn constructor() -> Self {
            Self {}
        }
    }

    impl TraitDefinition for Contract {
        #[ink(message)]
        fn message(&self) {}
    }
}

fn main() {}
