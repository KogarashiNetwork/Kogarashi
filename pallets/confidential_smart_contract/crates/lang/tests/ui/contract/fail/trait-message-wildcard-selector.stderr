error: encountered conflicting ink! attribute argument
 --> tests/ui/contract/fail/trait-message-wildcard-selector.rs:6:24
  |
6 |         #[ink(message, selector = _)]
  |                        ^^^^^^^^^^^^

error: wildcard selectors are only supported for inherent ink! messages or constructors, not for traits.
 --> tests/ui/contract/fail/trait-message-wildcard-selector.rs:6:24
  |
6 |         #[ink(message, selector = _)]
  |                        ^^^^^^^^^^^^

error[E0432]: unresolved import `super::foo::TraitDefinition`
  --> tests/ui/contract/fail/trait-message-wildcard-selector.rs:15:9
   |
15 |     use super::foo::TraitDefinition;
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `TraitDefinition` in `foo`
